### PR TITLE
chore: bump dependencies

### PR DESCRIPTION
**Pull request by capJavert**
Bump dependencies.

Some major upgrades but no breaking changes since packages just incremented major because of old node@14,16 support.

### CODE REVIEW

**Comment by capJavert** [comment]
I think we can also remove `node-fetch` as dependency since node 20.x supports fetch natively. I'll do it in separate PR.

File: package.json [hunk 1]
```diff
@@ -121,38 +121,38 @@
...
-    "@swc/core": "^1.4.14",
+    "@fastify/static": "^7.0.4",
+    "@swc/core": "^1.5.5",
     "@types/express": "^4.17.21",
     "@types/jest": "^29.5.12",
     "@types/jsonwebtoken": "^9.0.6",
-    "@types/lodash": "^4.17.0",
-    "@types/node": "^18.16.17",
+    "@types/lodash": "^4.17.1",
+    "@types/node": "^20.12.10",
```
Comment (omBratteng) [inline package.json:130]: I usually just do `20.x`, then it's taken care of for this version

```suggestion
    "@types/node": "20.x",
```

**Review by omBratteng — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (capJavert) [reply package.json:130]: I don't like wildcards 😆 they can bite you really easily

**Review by rebelchris — APPROVED** [review]
Nice nice!

File: package.json (hunk 1 above)
Comment (omBratteng) [reply package.json:130]: Same, but I usually find it alright for `@types/*`, as it's just types, and shouldn't cause issues™️

Comment (omBratteng) [reply package.json:130]: But non-blocker

**Pull request merged**